packages/provider/src/transports/extension-transport/extension-message-handler.ts

```diff
@@ -0,0 +1,29 @@
+import { WalletRequestHandler } from '../wallet-request-handler'
+import { BaseWalletTransport } from '../base-wallet-transport'
+import { InitState, ProviderMessage } from '../../types'
+
+export const CHANNEL_ID = 'sequence-extension-message-handler'
+
+export class ExtensionMessageHandler extends BaseWalletTransport {
+  private port: any
+
+  constructor(walletRequestHandler: WalletRequestHandler) {
+    super(walletRequestHandler)
+    this._init = InitState.OK
+  }
+
+  // from within extension execution context
+  register() {
+    this._registered = true
+
+    // TODO: check execution context, throw if not an extension
```

Comment:
good point on the TODO.. we should def check `if ((chrome as any) === undefined) { .. throw..}` or something like it